test/hummingbot/connector/exchange/bittrex/test_bittrex_order_book_data_source.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+import json
+import unittest
+from unittest.mock import AsyncMock, patch
+
+from hummingbot.connector.exchange.bittrex.bittrex_api_order_book_data_source import \
+    BittrexAPIOrderBookDataSource
+
+
+class BittrexOrderBookDataSourceTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.base_asset = "COINALPHA"
+        cls.quote_asset = "HBOT"
+        cls.trading_pair = f"{cls.base_asset}-{cls.quote_asset}"
+        cls.symbol = f"{cls.base_asset}{cls.quote_asset}"
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.ev_loop = asyncio.get_event_loop()
+
+        self.ws_incoming_messages = asyncio.Queue()
+        self.resume_test_event = asyncio.Event()
+        self._finalMessage = 'FinalDummyMessage'
+
+        self.output_queue = asyncio.Queue()
+
+        self.ob_data_source = BittrexAPIOrderBookDataSource(trading_pairs=[self.trading_pair])
+
+    def _create_queue_mock(self):
+        queue = AsyncMock()
+        queue.get.side_effect = self._get_next_ws_received_message
+        return queue
+
+    async def _get_next_ws_received_message(self):
+        message = await self.ws_incoming_messages.get()
+        if json.loads(message) == self._finalMessage:
+            self.resume_test_event.set()
+        return message
+
+    @patch("signalr_aio.Connection.start")
+    @patch("asyncio.Queue")
+    @patch(
+        "hummingbot.connector.exchange.bittrex.bittrex_api_order_book_data_source.BittrexAPIOrderBookDataSource"
+        "._transform_raw_message"
+    )
+    def test_listen_for_trades(self, transform_raw_message_mock, mocked_connection, _):
+        mocked_connection.return_value = self._create_queue_mock()
```

Comment:
Instead of patching asyncio.Queue I think it would be better to keep a reference to the real queue in the order book data source object and use the reference to put the messages we need for testing.

Reply:
The goal of this is to mock line 178 of `bittrex_api_order_book_data_source.py`:
```Python
    async def _checked_socket_stream(self, connection: signalr_aio.Connection) -> AsyncIterable[str]:
        try:
            while True:
                async with timeout(MESSAGE_TIMEOUT):  # Timeouts if not receiving any messages for 10 seconds(ping)
THIS =>             msg = await connection.msg_queue.get()
                    yield msg
        except asyncio.TimeoutError:
            self.logger().warning("Message queue get() timed out. Going to reconnect...")
```
The `mst_queue` is an `asyncio.Queue` used in `signalr_aio.Connection`